src/server/saved_objects/service/lib/repository.test.js

```diff
@@ -1502,4 +1512,38 @@ describe('SavedObjectsRepository', () => {
       }
     });
   });
+
+  describe('hidden types', () => {
+    it('should error when attempting to \'get\' hidden types', async () => {
+      await expect(savedObjectsRepository.get('hiddenType')).rejects.toEqual(new Error('Hidden types not allowed!'));
+    });
+
+    it('should error when attempting to \'bulkGet\' hidden types', async () => {
+      callAdminCluster.returns({ docs: [] });
+      await savedObjectsRepository.bulkGet([
+        { id: 'one', type: 'config' },
+        { id: 'two', type: 'index-pattern' },
+        { id: 'three', type: 'globaltype' },
+        { id: 'four', type: 'hiddenType' },
+      ]);
+      sinon.assert.calledWithExactly(callAdminCluster, 'mget', {
+        body: {
+          docs: [
+            { _id: 'config:one', _type: 'doc' },
+            { _id: 'index-pattern:two', _type: 'doc' },
+            { _id: 'globaltype:three', _type: 'doc' },
+          ]
+        },
+        index: '.kibana-test',
+      });
+    });
+
+    it('should error when attempting to \'find\' hidden types', async () => {
+      await expect(savedObjectsRepository.find({ type: 'hiddenType' })).rejects.toEqual(new Error('Hidden types not allowed!'));
```

Comment:
Can we add tests for when `type` is an array?

Reply:
Tests added!